test/helpers/kubectl.go

```diff
@@ -2791,9 +2793,24 @@ func addrsEqual(addr1, addr2 *models.BackendAddress) bool {
 
 // GenerateNamespaceForTest generates a namespace based off of the current test
 // which is running.
+// Note: Namespaces can only be 63 characters long (to comply with DNS). We
+// ensure that the namespace here is shorter than that, but keep it unique by
+// hashing the complete name.
 func GenerateNamespaceForTest() string {
 	lowered := strings.ToLower(ginkgoext.CurrentGinkgoTestDescription().FullTestText)
 	// K8s namespaces cannot have spaces.
 	replaced := strings.Replace(lowered, " ", "", -1)
-	return replaced
+
```

Comment:
btw, there is a way to get the test name in a better way: 
https://github.com/cilium/cilium/blob/master/test/ginkgo-ext/scopes.go#L145

Reply:
👍 Thanks eloy! I switched it out to use `GetTestName`

Reply:
ah, we can't have `_` in the namespaces... back we go I guess